### ISSUE TITLE
Zip/Unzip tasks

### ISSUE THREAD

**Issue by Rarst** [Codegyre/Robo#176]
While PHP usually got `ZipArchive` available, it is kind of a kludge to use. Like these things go the API is tad too low level. [Ends up messy among neat Robo task calls](https://github.com/Rarst/laps/blob/a3572063d293b09d942df9a66371c45fdbcfb052/RoboFile.php#L135-L154).

Zip/Unzip tasks would make sense to me, with dependency on `ZipArchive` being available in PHP and Symfony Finder (which is already in Robo's dependencies).

Can create them and PR if there is interest. :)

**Comment by burzum** [comment]
Sounds good to me but this might raise a question if it wouldn't make more sense to directly implement a wrapper around Gaufrette or Flysystem. They abstract storage systems pretty well and even some archives. So the interface stays the same, no matter if you put a file on a FTP or into an archive.

**Comment by Rarst** [comment]
What would the scope and task API be then?

I think "zip into file" or "unzip from file" is pretty succinct tasks. "manipulate filesystem abstraction" seems to get a little too fuzzy.

**Comment by roland-d** [comment]
Interesting, I just wrote a zip task for myself as my first lesson on how to use Robo.li. I actually followed the structure of the PackPhar task. Since it is basically the same thing I thought it would make sense to try and keep the same structure.

The one thing I did do different in my task is, I supplied a folder and it packs the folder and files into the supplied zip name. I didn't want to call the addFile function 415 times in my case.

Ideally in my opinion we have a zip and unzip function. The addFile function should take both folders and files as argument. Even better would be to be able to supply an alternate path in the zip. Going to stop here before my wishlist shows up :)

**Comment by Rarst** [comment]
> The addFile function should take both folders and files as argument. Even better would be to be able to supply an alternate path in the zip.

Yeah, that's about same as I have in mind:

1. Zip file/folder, optionally into custom path, into zip file.
2. Extract all off / specific folder/file from zip file.

**Comment by DavertMik** [comment]
@roland-d could you send a Pull Request with your task? I agree ZIP task would be really handy

**Comment by roland-d** [comment]
@DavertMik I am going to work on the PR. My zipcode is currently not a single task, so I will move it there and I will need to setup Robo using composer and not the phar I have now. Give me a bit of time and I will get it done.

**Comment by boedah** [comment]
:+1: 

@roland-d Please support Symfony Finder (http://symfony.com/doc/current/components/finder.html), so one can construct a Finder object beforehand and pass it on to the ZipTask?
This would basically mean:

* supporting `IteratorAggregate`
* supporting `Iterator`
* supporting `Traversable` and arrays

Have a look at the `append()` method of Finder, which also deals with all of them.

Another idea:

Make an abstract base task named `Compress`, which handles all the file params (input, compressed file name) and a subtask `Zip` which only contains the (minimum) methods to compress a ZIP?
So other methods can be easily plugged into later on.

What do you think?

**Comment by DavertMik** [comment]
@roland-d thanks! will be waiting for it

@boedah I think that something like PackPhar will work. You can use it with Symfony Finder or specify each files manually. Free choice )

**Comment by roland-d** [comment]
@DavertMik I have created the PR #189 but I have been unable to test it myself in the new setup. If I setup Robo.li with composer certain tasks simply don't work. For example PackPhar gives me this error:

> Fatal error: Call to undefined method RoboFile::PackPhar()

While the task SemVer works without an issue. I have tested this with a RoboFile.php with only this code:
```php
<?php
/**
 * This is project's console commands configuration for Robo task runner.
 *
 * @see http://robo.li/
 */
class RoboFile extends \Robo\Tasks
{
    // define public methods as commands

	public function release()
	{
		$pharTask = $this->PackPhar('package/codecept.phar')
						 ->compress()
						 ->stub('package/stub.php');
	}
}
```

Not sure what is going on :-|

**Comment by greg-1-anderson** [comment]
I believe this is fixed by #244, once the issue with the tests is worked out.

**Comment by greg-1-anderson** [comment]
#244 has been merged.
